grpc/cmd/server/main.go

```diff
@@ -29,11 +29,26 @@ func main() {
 	}
 	defer lis.Close()
 
-	go runGRPC(lis)
-	runHTTP(clientAddr)
+	//this channel will be used to halt
```

Comment:
IMHO it's less readable. Having `log.Fatal(..)` makes `runGRPC` and `runHTTP` self-sufficient and independent. No need to involve channels for such simple use case.